### PR TITLE
Feature/plcn 350 create create admin command handler

### PR DESCRIPTION
**Pull request by AntonEsbjergIT-Minds**
Created a commandhandler for CreateAdminCommand, the Admin is created from the request, but the password is hashed first to encrypt it. Tests have also been implemented

### CODE REVIEW

**Review by itm-mne — APPROVED** [review]

File: src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	public async Task<Result> Handle(CreateAdminCommand request, CancellationToken cancellationToken)
+	{
+		var userEntity = new AdminUser
+		{
+			Email = request.Email,
+			Password = _passwordHasher.Hash(request.Password),
+			Name = request.Name,
+		};
+
+		await _unitOfWork.UserRepository.CreateAsync(userEntity);
```
Comment (Nicolai-Bak) [inline src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs:30]: Should there be any check on whether the email and name already exists before creating a new admin?

File: src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using Pelican.Application.Abstractions.Authentication;
+using Pelican.Application.Abstractions.Data.Repositories;
+using Pelican.Application.Abstractions.Messaging;
+using Pelican.Application.Users.Commands.CreateAdmin;
+using Pelican.Domain.Entities.Users;
+using Pelican.Domain.Shared;
+
+namespace Pelican.Application.Authentication.CreateAdmin;
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs:8]: Moved it to that namespace and folder

File: test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	public CreateAdminCommandHandlerTests()
+	{
+		_uut = new CreateAdminCommandHandler(_unitOfWorkMock.Object, _passwordHasherMock.Object);
+	}
+
+	[Fact]
+	public void Constructor_UnitOfWorkNull_ArgumentNullException()
+	{
+		//Arrange
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs:26]: Removed

File: test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+		Assert.Contains(
+			"unitOfWork",
+			result.Message);
+	}
+
+	[Fact]
+	public void Constructor_PassWordHasherNull_ArgumentNullException()
+	{
+		//Arrange
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs:42]: Removed

File: src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs (hunk 1 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs]: Created a check on email as per our conversation.

File: src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	private readonly IPasswordHasher _passwordHasher;
+
+	public CreateAdminCommandHandler(
+		IUnitOfWork unitOfWork,
+		IPasswordHasher passwordHasher)
+	{
+		_unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
+		_passwordHasher = passwordHasher ?? throw new ArgumentNullException(nameof(passwordHasher));
+	}
+	public async Task<Result> Handle(CreateAdminCommand request, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Application/Authentication/CreateAdmin/CreateAdminCommandHandler.cs]: I have made the change

File: test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+		_unitOfWorkMock
+			.Verify(x => x
+				.UserRepository
+				.CreateAsync(expectedUser, default), Times.Once);
+
+		_unitOfWorkMock
+			.Verify(x => x
+				.SaveAsync(default), Times.Once);
+
+		_passwordHasherMock.Verify(x => x.Hash("test"));
```
Comment (Nicolai-Bak) [inline test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs:132]: if you want to verify the method is called with correct param I would not user "test" for all input.

File: test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+using Pelican.Application.Abstractions.Authentication;
+using Pelican.Application.Abstractions.Data.Repositories;
+using Pelican.Application.Abstractions.Messaging;
+using Pelican.Application.Users.Commands.CreateAdmin;
+using Pelican.Domain.Entities;
+using Pelican.Domain.Entities.Users;
+using Xunit;
+
+namespace Pelican.Application.Test.CreateAdmin.Commands;
+public class CreateAdminCommandHandlerTests
```
Comment (Nicolai-Bak) [inline test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs:12]: Use of `Times.` should be consistent throughout verifies. 

Indentation style differs a bit on dependency setup and verify.

File: test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs (hunk 6 above)
Comment (AntonEsbjergIT-Minds) [reply test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs:132]: Now using naming as so "test{Param}".

File: test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs (hunk 7 above)
Comment (AntonEsbjergIT-Minds) [reply test/Pelican.Application.Test/CreateAdmin/Commands/CreateAdminCommandHandlerTests.cs:12]: Fixed indentation and using Times.Once now

**Review by s-bogelund — APPROVED** [review]

**Review by Nicolai-Bak — APPROVED** [review]

**Pull request merged**